zipline/gens/brokers/alpaca_broker.py

```diff
@@ -0,0 +1,261 @@
+
+    def is_alive(self):
+        try:
+            self._api.get_account()
+            return True
+        except BaseException:
+            return False
+
+    def _order2zp(self, order):
+        zp_order = ZPOrder(
+            id=order.client_order_id,
+            asset=symbol_lookup(order.symbol),
+            amount=int(order.qty) if order.side == 'buy' else -int(order.qty),
+            stop=float(order.stop_price) if order.stop_price else None,
+            limit=float(order.limit_price) if order.limit_price else None,
+            dt=order.submitted_at,
+            commission=0,
+        )
+        zp_order.status = ZP_ORDER_STATUS.OPEN
+        if order.canceled_at:
+            zp_order.status = ZP_ORDER_STATUS.CANCELLED
+        if order.failed_at:
+            zp_order.status = ZP_ORDER_STATUS.REJECTED
+        if order.filled_at:
+            zp_order.status = ZP_ORDER_STATUS.FILLED
+            zp_order.filled = int(order.filled_qty)
+        return zp_order
+
+    def _new_order_id(self):
+        return uuid.uuid4().hex
+
+    def order(self, asset, amount, style):
+        symbol = asset.symbol
+        qty = amount if amount > 0 else -amount
+        side = 'buy' if amount > 0 else 'sell'
+        order_type = 'market'
+        if isinstance(style, MarketOrder):
+            order_type = 'market'
+        elif isinstance(style, LimitOrder):
+            order_type = 'limit'
+        elif isinstance(style, StopOrder):
+            order_type = 'stop'
+        elif isinstance(style, StopLimitOrder):
+            order_type = 'stop_limit'
+
+        limit_price = style.get_limit_price(side == 'buy') or None
+        stop_price = style.get_stop_price(side == 'buy') or None
+
+        zp_order_id = self._new_order_id()
+        dt = pd.to_datetime('now', utc=True)
+        zp_order = ZPOrder(
+            dt=dt,
+            asset=asset,
+            amount=amount,
+            stop=stop_price,
+            limit=limit_price,
+            id=zp_order_id,
+        )
+
+        order = self._api.submit_order(
+            symbol=symbol,
+            qty=qty,
+            side=side,
+            type=order_type,
+            time_in_force='day',
+            limit_price=limit_price,
+            stop_price=stop_price,
+            client_order_id=zp_order.id,
+        )
+        zp_order = self._order2zp(order)
+        return zp_order
+
+    @property
+    def orders(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders()
+        }
+
+    @property
+    def transactions(self):
+        return {
+            o.client_order_id: self._order2zp(o)
+            for o in self._api.list_orders(status='all')
+        }
+
+    def cancel_order(self, zp_order_id):
+        try:
+            order = self._api.get_order_by_client_order_id(zp_order_id)
+            self._api.cancel_order(order.id)
+        except Exception as e:
```

Comment:
Can we use more specific exception type here?

Reply:
Sure